File: sc_status.php
<?php
////////////////////////////////////////////////////////////////////////////////
// script name: sc status
// date: 10/01/2012
// author: davestj@gmail.com
// cause: checks status of a shoutcast server and display's online or offline status
// version: 1.1 b
// platform independent
// file name: sc_status.php
////////////////////////////////////////////////////////////////////////////////
//do not get cute and monkey with anything below, unless you know what u are doing.
include ('./config.php');

// override ini settings for script execution time, we dont need a minute to decide
// if a server is up or not, 10 seconds should be sufficient.
ini_set("max_execution_time", "10");

//check config settings
if($useimage == 'yes' && $usetext == 'yes'){
	echo 'You must choose text display or image display but not both<br>
		 please edit your config.php file<br>';
	exit();
}
//lets initiate a tcp socket connection to determine whether or not the server
//is actually up.
$scp = @fsockopen($sc_ip, $sc_port, $errno, $errstr, 30);
//let me know where or not its up
	if(!$scp){
	    $sock_init = 'FALSE';
	}

//show them whether or not the server is actually up or not
	if($sock_init == 'FALSE'){
		if($useimage == 'yes'){
		    echo ''.$station_name.' <img srv='.$offline_imgurl.'>';
		}else if ($usetext == 'yes'){
		    echo ''.$station_name.' - '.$offline_text.'';
		}
	}
//check 7.html to see if dsp is connected
	if($sock_init != 'FALSE'){
         fputs($scp,"GET /7.html HTTP/1.0\r\nUser-Agent: SC Status (Mozilla Compatible)\r\n\r\n");
 		while(!feof($scp)) {
  			$sc7 .= fgets($scp, 1024);
 			}
//close it up
@fclose($scp);
//while we got the page open into memory lets bomb n parse baby.
$sc7 = preg_replace('/^\<body\>/', "", $sc7);
$sc7 = preg_replace('/^\<\/body\>/', ",", $sc7);
$sc_contents = explode(",",$sc7);
$dummy = $sc_contents[0];
$dsp_connected = $sc_contents[1];


//check dsp connection and display the status of the shoutcast server in question
//do images first
 	if($sock_init != 'FALSE'){
 	    if($dsp_connected == '1' && $useimage == 'yes'){
  		echo ''.$station_name.' <img srv='.$online_imgurl.'>';
 	}else if ($dsp_connected != '1' && $useimage == 'yes'){
        echo ''.$station_name.' <img srv='.$offline_imgurl.'>offline image';
	 }
}
 //do text if set
if($sock_init != 'FALSE'){
	if ($dsp_connected == '1' && $usetext == 'yes'){
		echo ''.$station_name.' - '.$online_text.'';
 	}else if($dsp_connected != '1' && $usetext == 'yes'){
 	    echo ''.$station_name.' - '.$offline_text.'';
 	}
  }

}//end 7.html
	
	
	




//EOF
?>

File: sc_status.py
"""
@package sc status
@date 8/20/2010
@author davestj@gmail.com
@description checks status of a shoutcast server and display's online or offline status
@version 1.0 final
@abstract Procedural script to get data from Shoutcast D.N.A.S
@filename sc_status.py
"""

import socket

# Config settings
from config import sc_ip, sc_port, useimage, usetext, station_name, offline_imgurl, offline_text, online_imgurl, online_text

# Override ini settings for script execution time, we don't need a minute to decide
# if a server is up or not, 10 seconds should be sufficient.
import sys
sys.setrecursionlimit(10)

# Check config settings
if useimage == 'yes' and usetext == 'yes':
    print('You must choose text display or image display but not both\n'
          'please edit your config.py file')
    exit()

# Let's initiate a TCP socket connection to determine whether or not the server
# is actually up.
scp = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
scp.settimeout(10)  # Set socket timeout to 10 seconds

# Let me know whether or not it's up
try:
    scp.connect((sc_ip, sc_port))
    sock_init = True
except (socket.timeout, ConnectionRefusedError):
    sock_init = False

# Show them whether or not the server is actually up or not
if not sock_init:
    if useimage == 'yes':
        print(f'{station_name} <img srv={offline_imgurl}>')
    elif usetext == 'yes':
        print(f'{station_name} - {offline_text}')
else:
    # Check 7.html to see if DSP is connected
    scp.send(b'GET /7.html HTTP/1.0\r\nUser-Agent: SC Status (Mozilla Compatible)\r\n\r\n')
    sc7 = scp.recv(1024).decode()
    scp.close()

    # While we got the page open into memory let's parse it.
    sc7 = sc7.replace('<body>', '').replace('</body>', ',')
    sc_contents = sc7.split(',')
    dummy = sc_contents[0]
    dsp_connected = sc_contents[1]

    # Check DSP connection and display the status of the Shoutcast server in question
    # Do images first
    if dsp_connected == '1' and useimage == 'yes':
        print(f'{station_name} <img srv={online_imgurl}>')
    elif dsp_connected != '1' and useimage == 'yes':
        print(f'{station_name} <img srv={offline_imgurl}> offline image')

    # Do text if set
    if dsp_connected == '1' and usetext == 'yes':
        print(f'{station_name} - {online_text}')
    elif dsp_connected != '1' and usetext == 'yes':
        print(f'{station_name} - {offline_text}')
